oxart/devices/mqtt_client/driver.py

```diff
@@ -0,0 +1,149 @@
+"""
+Simple driver for sending MQTT commands to a Sinara stabilizer or a mezzanine board on
+it. Requires Quartiq miniconf-mqtt package from https://github.com/quartiq/miniconf.
+"""
+
+from miniconf import Miniconf
+import asyncio
+import logging
+from typing import Any
+import nest_asyncio
+import argparse
+
+nest_asyncio.apply()
+
+logger = logging.getLogger()
+
+
+class MQTTClient:
+
+    def __init__(self, dmgr, prefix: str, address: str, timeout: float = 1):
+        """
+        :param dmgr: Device manager passed in by Artiq.
+        :param prefix: Prefix to the '/settings' topic for the device.
+        :param address: IP adress of the device running the MQTT broker.
+        """
+
+        self.dmgr = dmgr
```

Comment:
Why are we passing in a dmgr?

Reply:
When instantiating the object from the HasEnvironment object, the first argument passed is a dmgr (though I'm unsure why and whether this is configurable behaviour). This is despite the fact that there is no explicit "dmgr" argument defined in the device_db. I haven't found references to this in the documentation, but printing the dmgr shows it is indeed a device manager (and not having it as first argument causes errors, since then the "prefix" first argument is passed to twice).

If there is a way to not pass a dmgr when instantiating from the environment, this would be better.

Reply:
Ok, I think if we can not pass it on the user side this would be better. Since this object never uses the `dmgr`. In fact, in our use case we wrap this object anyway, so could just not pass in the `dmgr`. Doesn't hurt either way, I suppose. Though most drivers in this repo don't have the dmgr  since they're intended to be used as controllers.

Reply:
After discussion, this doesn't interface with any hardware over usb serial etc. So we can't see any reason to run as a controller over a local instance on an artiq master. As such, it makes the most sense to keep the `dmgr` so it can be used easily as a local device. 
If you do need a controller, `None` will do as the `dmgr`